skfuzzy/control/controlsystem.py

```diff
@@ -428,19 +428,40 @@ def find_memberships(self):
         if len(self.var.terms.keys()) == 0:
             raise ValueError("Set term membership function(s) first")
 
+        '''
+        First we have to upsample the universe of self.var in order to add the key points of the membership function
+        based on the activation level for this consequent, using the interp_value function, which interpolates the `xx`
+        values in the universe such that its membership function value is the activation level.
+        '''
+        add_universe = set()
+        for label, term in self.var.terms.items():
+            cut = term.membership_value[self.sim]
+            if cut is None:
+                continue  # No membership defined for this adjective
+            add_xx = interp_value(self.var.universe, term.mf, cut)
+            add_universe.update(add_xx)
+        # We are only interested in points not in self.var.universe
+        add_universe = add_universe-set(self.var.universe)
+        upsampled_universe = zip(self.var.universe.tolist() + list(add_universe), range(len(self.var.universe))+[None]*len(add_universe))
```

Comment:
This line is throwing the test suite error due to a change in how `range` works in Python 3 - it returns a generator instead of a list.

If you want backwards-compatible `range`, then `from past.builtins import range` at the top and use `range`. For forwards-compatible behavior (recommended), tweak the loop to work with `range` obtained from the import `from builtins import range`.